Xamarin.PropertyEditing.Mac/Controls/DateEditorControl.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using AppKit;
+using Xamarin.PropertyEditing.Common;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	internal class DateEditorControl : BaseDateTimeEditorControl<Date>
+	{
+
+		public DateEditorControl (IHostResourceProvider hostResources)
+			: base (hostResources)
+		{
+			DatePicker.DatePickerElements = NSDatePickerElementFlags.YearMonthDateDay;
+		}
+
+		protected override void Editor_Activated (object sender, EventArgs e)
+		{
+			ViewModel.Value = new Date (DatePicker.DateValue.ToDateTime ());
```

Comment:
It would be great add some try/catch block here.

Reply:
Actually, I don't agree here, `DatePicker.DateValue` should always return a value, but I do not know what the behavior of `ToDateTime()`, I am making an assumption that it will always return a valid value for the `Date` ctor, if that is not true, then we should be validating that value before constructing the `Date`.

The Event processor for `Editor_Activated` should be handling any exceptions thrown by event handlers so that all event handlers are called properly.